service/header/core_exchange.go

```diff
@@ -31,6 +28,20 @@ func NewCoreExchange(fetcher *core.BlockFetcher, dag format.DAGService) *CoreExc
 	}
 }
 
+// Start starts the CoreExchange by ensuring its core client
+// is running.
+func (ce *CoreExchange) Start(context.Context) error {
+	// ensure the core client is running
+	if !ce.fetcher.IsRunning() {
+		return fmt.Errorf("client not running")
+	}
```

Comment:
If `fetcher core.BlockFetcher` is started with DI then DI guarantees that fetcher IsRunning, so this check might be useless.

Reply:
What if it dies after start-up? How does DI catch that?

Reply:
What if P2PExchange dies after the start and Syncer does not catch this?

Reply:
And there are other similar cases, I am just not sure if we need to track if the component you depend on is started or else. IMHO, we should aim for `Start` returning nil guarantee if everything is fine. If not, the error will be returned by accessing the method of started but failed component.

Reply:
Also, how fetcher can die? Don't know

Reply:
Okay that's fair. I'll remove the `IsStarted` method